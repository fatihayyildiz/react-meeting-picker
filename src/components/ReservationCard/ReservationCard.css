.clear-link {
	text-decoration: underline;
	color: darkcyan;
	font-size: 10px;
	cursor: pointer;
}
