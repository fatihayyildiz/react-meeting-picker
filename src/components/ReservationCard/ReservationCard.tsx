import React from 'react';

export type ReservationCardProps = {
	day: string;
	start_time: string;
	end_time: string;
};

const ReservationCard = ({
	day,
	start_time,
	end_time,
}: ReservationCardProps) => {
	return (
		<div className="container">
			<div className="row">
				<div className="col-12">Reservation</div>
			</div>
			<div className="row">
				<div className="col-12">{day}</div>
			</div>
			<div className="row">
				<div className="col-5">{start_time}</div>
				<div className="col-2">-</div>
				<div className="col-5">{end_time}</div>
			</div>
		</div>
	);
};

export default ReservationCard;
