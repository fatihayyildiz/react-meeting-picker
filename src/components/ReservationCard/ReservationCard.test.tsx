import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import ReservationCard from 'components/ReservationCard/ReservationCard';

afterEach(cleanup);

const testProps = {
	day: 'July 9, 2018',
	start_time: '08:00',
	end_time: '09:00',
};

test('Reservation card should match with snapshot', () => {
	// First render component and get as fragment
	const { asFragment } = render(
		<ReservationCard
			{...{
				day: testProps.day,
				start_time: testProps.start_time,
				end_time: testProps.end_time,
			}}
		/>
	);

	// Compare with saved snapshot
	expect(asFragment()).toMatchSnapshot();
});

test('Reservation card render with props should match with snapshot', () => {
	// First render component with props and get as fragment
	const { asFragment } = render(
		<ReservationCard
			{...{
				day: testProps.day,
				start_time: testProps.start_time,
				end_time: testProps.end_time,
			}}
		/>
	);

	// Compare with saved snapshot
	expect(asFragment()).toMatchSnapshot();
});

test('Reservation card renders times correctly', () => {
	// First render component with props and get as fragment
	render(
		<ReservationCard
			{...{
				day: testProps.day,
				start_time: testProps.start_time,
				end_time: testProps.end_time,
			}}
		/>
	);

	// Compare start time
	expect(screen.getByText(testProps.day)).toHaveTextContent(testProps.day);

	// Compare start time
	expect(screen.getByText(testProps.start_time)).toHaveTextContent(
		testProps.start_time
	);

	// Compare end time
	expect(screen.getByText(testProps.end_time)).toHaveTextContent(
		testProps.end_time
	);
});
