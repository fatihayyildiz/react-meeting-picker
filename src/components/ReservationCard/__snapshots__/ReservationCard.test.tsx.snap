// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation card should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-9"
      >
        Reservation
      </div>
      <div
        class="col-3 my-auto"
      >
        <span
          class="clear-link"
        >
          clear
        </span>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-12"
      >
        July 9, 2018
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        08:00
      </div>
      <div
        class="col-2"
      >
        -
      </div>
      <div
        class="col-5"
      >
        09:00
      </div>
    </div>
  </div>
</DocumentFragment>
`;
